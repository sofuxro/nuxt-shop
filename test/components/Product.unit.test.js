import { factoryMount } from '~/test/utils/factories'

import Component from '~/components/Product'

let wrapper
const store = {
    modules: {
        products: {
            namespaced: true,
            getters: {
                getQuantity: (state) => (id) => 1
            },
            actions: {
                addToCart: jest.fn()
            }
        }
    }
}

beforeEach(() => {
    wrapper = factoryMount(Component, { store })
})

afterEach(() => {
    wrapper.destroy()
})

describe('Product', () => {
    test('has default item prop', () => {
        expect(wrapper.vm.item).toBeTruthy()
    })

    test('gender showing the correct text', () => {
        // Male
        expect(wrapper.element).toMatchSnapshot()

        // Female
        wrapper.setProps({
            item: { ...wrapper.vm.item, tags: { gender: 'female' } }
        })
        expect(wrapper.element).toMatchSnapshot()
    })

    test('add to cart', () => {
        const button = wrapper.find('button')
        button.trigger('click')
        expect(store.modules.products.actions.addToCart).toHaveBeenCalled()
    })
})
