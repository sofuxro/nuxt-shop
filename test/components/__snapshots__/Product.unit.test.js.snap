// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product gender showing the correct text 1`] = `
<a
  class="product-in-list"
  to="/products/-1"
>
  <div
    class="v-responsive v-image"
  >
    <div
      class="v-responsive__content"
    />
  </div>
   
  <div
    class="product-in-list__gender"
  >
    MENS WATCHES
  </div>
   
  <div>
    No Name
  </div>
   
  <div
    class="product-in-list__product_action"
  >
    <div
      class="price"
    >
      -$1.00
    </div>
     
    <button>
      ADD TO CART
    </button>
  </div>
</a>
`;

exports[`Product gender showing the correct text 2`] = `
<a
  class="product-in-list"
  to="/products/-1"
>
  <div
    class="v-responsive v-image"
  >
    <div
      class="v-responsive__content"
    />
  </div>
   
  <div
    class="product-in-list__gender"
  >
    MENS WATCHES
  </div>
   
  <div>
    No Name
  </div>
   
  <div
    class="product-in-list__product_action"
  >
    <div
      class="price"
    >
      -$1.00
    </div>
     
    <button>
      ADD TO CART
    </button>
  </div>
</a>
`;
